From 7807bcc55b4927fc327830d2237200772d2e1106 Mon Sep 17 00:00:00 2001
From: Rafael Guterres Jeffman <rjeffman@redhat.com>
Date: Fri, 17 Jun 2022 15:40:04 -0300
Subject: [PATCH] webui IdP: Remove arrow notation due to uglify-js limitation.

uglify-js 2.x series do not support ECMAScript 6 arrow notation ('=>')
for callback definition.

This patch changes the arrow definition callbacks for regular anonymous
function definitions.
---
 install/ui/src/freeipa/idp.js | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/install/ui/src/freeipa/idp.js b/install/ui/src/freeipa/idp.js
index ada09c075..be3c4f0e6 100644
--- a/install/ui/src/freeipa/idp.js
+++ b/install/ui/src/freeipa/idp.js
@@ -227,7 +227,7 @@ IPA.add_idp_policy = function() {
         // For custom template we show custom fields
         // and mark all of them required and passed to the RPC
         // If show_custom is false, the opposite happens
-        custom_fields.forEach(fname => {
+        custom_fields.forEach(function(fname) {
             widget_f = that.container.fields.get_field(fname);
             widget_f.set_required(show_custom);
             widget_f.set_enabled(show_custom);
@@ -235,7 +235,7 @@ IPA.add_idp_policy = function() {
         });
 
         // For template fields we show them if custom aren't shown
-        template_fields.forEach(fname => {
+        template_fields.forEach(function(fname) {
             widget_f = that.container.fields.get_field(fname);
             widget_f.set_enabled(!show_custom);
             widget_f.widget.set_visible(!show_custom);
@@ -252,7 +252,7 @@ IPA.add_idp_policy = function() {
         var value = prov_f.get_value()[0];
 
         // First, clear template fields from the previous provider choice
-        template_fields.forEach(fname => {
+        template_fields.forEach(function(fname) {
             widget_f = that.container.fields.get_field(fname);
             widget_f.widget.set_visible(false);
             widget_f.set_required(false);
@@ -260,9 +260,9 @@ IPA.add_idp_policy = function() {
         });
 
         // Second, enable and get required template-specific fields
-        idp.templates.forEach(idp_v => {
+        idp.templates.forEach(function(idp_v) {
             if (idp_v['value'] == value) {
-                idp_v['fields'].forEach(fname => {
+                idp_v['fields'].forEach(function(fname) {
                     widget_f = that.container.fields.get_field(fname);
                     widget_f.set_required(true);
                     widget_f.set_enabled(true);
-- 
2.36.1

